File: WeatherServer/WeatherServer.API/Models/Location.cs
using System.Text.Json.Serialization;

namespace WeatherServer.API.Models
{
    public record Location
        (
            string name,
            string region,
            string country,
            double lat,
            double lon,
            [property: JsonPropertyName("tz_id")] string tzID,
            [property: JsonPropertyName("localtime_epoch")] int localtimeEpoch,
            string localtime
        );
}

File: WeatherFlutter/weather_flutter/lib/model/location.dart
import 'package:json_annotation/json_annotation.dart';

part 'location.g.dart';

@JsonSerializable()
class Location {
  final String name;
  final String region;
  final String country;
  final double lat;
  final double lon;
  final String tzId;
  final int localTimeEpoch;
  final String localTime;

  Location({
    required this.name,
    required this.region,
    required this.country,
    required this.lat,
    required this.lon,

    @JsonKey(name: 'tz_id') required this.tzId,

    @JsonKey(name: 'localtime_epoch') required this.localTimeEpoch,

    @JsonKey(name: 'local_time') required this.localTime,
  });

  factory Location.fromJson(Map<String, dynamic> json) => _$LocationFromJson(json);
}
